autogen/agentchat/contrib/agent_builder.py

```diff
@@ -0,0 +1,428 @@
+import autogen
+import time
+import subprocess as sp
+import socket
+import os
+import json
+import hashlib
+from typing import Optional, List, Dict, Tuple
+from autogen.agentchat.contrib.gpt_assistant_agent import GPTAssistantAgent
+
+
+class AgentBuilder:
```

Comment:
Please add tests under test/contrib and add them to the contrib-test/contrib-openai-test workflows. @rickyloynd-microsoft @thinkall @BeibinLi @kevin666aa have done that before.